src/core/Fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,70 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/fieldset.css';
+
+export type FieldsetProps = {
+  /**
+   * The caption or title for the fieldset
+   */
+  legend?: React.ReactNode;
+} & React.ComponentPropsWithoutRef<'fieldset'>;
+
+/**
+ * Basic fieldset component to group different components together
+ * @example
+ * <Fieldset legend='General Settings' {...args}>
```

Comment:
It is a bit too big example for such component 😃 Maybe use this one from CSS:
![image](https://user-images.githubusercontent.com/36186912/128823689-49291fbc-4a94-423b-99cb-64eba22c04d8.png)
Of course without disabled state.

Reply:
I was thinking the same 😄 

And maybe we don't even need to specify optional props. So:
```jsx
<Fieldset legend='Settings'>
  <Input />
  <InputGroup>
    <ToggleSwitch />
    <ToggleSwitch />
  </InputGroup>
</Fieldset>
```